languages/ruby/lib/oso/polar/polar.rb

```diff
@@ -88,6 +88,49 @@ def self.join(separator, left, right)
         ffi_polar.validate_roles_config(validation_query_results)
       end
 
+      def get_allowed_resources(actor, action, klass) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+        resource = Variable.new 'resource'
+
+        # FIXME(gw) kind of a hack because of class autoreloading, maybe not needed?
+        if host.types.key? klass
+          class_name = host.types[klass].name
+        elsif host.types.key? klass.name
+          class_name = host.types[klass.name].name
+        else
+          raise "Unknown class `#{klass}`"
+        end
+
+        constraint = Expression.new(
+          'And',
```

Comment:
Why does this need to be an `And` with a single argument?

Reply:
that's just kind of the format the vm uses internally for partials right now